cs-json/src/main/java/io/cloudslang/content/json/utils/JsonUtils.java

```diff
@@ -90,18 +97,41 @@ private static void checkForNullValue(String actionString, String value) throws
         }
     }
 
-    public static JsonReader getJsonReader(String jsonObject, final AbstractJsonProvider provider) {
-        final Configuration configuration = Configuration.defaultConfiguration().jsonProvider(provider);
-        JsonReader jsonReader = new JsonReader(configuration);
-        jsonReader.parse(jsonObject);
-        return jsonReader;
+    public static JsonPath getValidJsonPath(final String jsonPath) {
+        try {
+            return JsonPath.compile(jsonPath);
+        } catch (IllegalArgumentException iae) {
+            throw illegalArgumentExceptionWithMessage(INVALID_JSONPATH, iae);
```

Comment:
Change the method name to a verb. Something like slamDunkIllegalArgumentExceptionWithMessage :smirk: